task_sdk/src/airflow/sdk/execution_time/context.py

```diff
@@ -315,3 +316,22 @@ def set_current_context(context: Context) -> Generator[Context, None, None]:
                 expected=context,
                 got=expected_state,
             )
+
+
+def context_update_for_unmapped(context: Context, task: BaseOperator) -> None:
+    """
+    Update context after task unmapping.
+
+    Since ``get_template_context()`` is called before unmapping, the context
+    contains information about the mapped task. We need to do some in-place
+    updates to ensure the template context reflects the unmapped task instead.
+
+    :meta private:
+    """
+    # TODO: Task-SDK this need to live in sdk too
+    from airflow.models.param import process_params
```

Comment:
Yeah we have a ticket for this. Can be tracked by https://github.com/apache/airflow/issues/44361